interview-questions.md

```diff
@@ -8,62 +8,62 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a parameter and an argument?
 
-  Your answer:
+  Your answer: A parameter is that is used to invoke a function. A argument is what is passed into the parameter and is ran in the function.
 
-  Researched answer:
+  Researched answer: A parameter is what is passed inside the function const  function = (parameter) => {} and an argument is passed to invoke the function console.log(function(argument))
 
 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
+  Your answer: There is three parameter in .map which are value, element, array. The optional ones are element and array.
 
-  Researched answer:
+  Researched answer: The parameter in .map which are value, element, array, callbackFn, and thisArg. The optional ones are element, array, callbackFn, and thisArg.
 
 
 
 3. What is the difference between map and filter?
 
-  Your answer:
+  Your answer: .map will return an array with the same length. .filter will make a choice and return an array with less than the original length.
 
-  Researched answer:
+  Researched answer: .map will return an array with each value converted. .filter will return an array with selected values from the original array.
 
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
+  Your answer: A function is a reuseable code that takes in a parameters and arguments to return an output. A method is a function inside of an object.
 
-  Researched answer:
+  Researched answer: A function is a reuseable code that takes in any parameters and arguments to return an output and can be dynamic. A method is a function inside of an object and is specific to that object.
 
 
 
 5. What is object destructuring?
 
-  Your answer:
+  Your answer: Object destructuring creates a shortcut to access date in an object.
 
-  Researched answer:
+  Researched answer: Object destructuring allows you to create code to refer back to an object and pull out specific data.
```

Comment:
Object destructuring allows you to assign a variable to a key within the object.